com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/SetBreakpointsRequestHandler.java

```diff
@@ -35,16 +41,31 @@
 import com.microsoft.java.debug.core.protocol.Responses;
 import com.microsoft.java.debug.core.protocol.Types;
 
-public class SetBreakpointsRequestHandler implements IDebugRequestHandler {
+public class SetBreakpointsRequestHandler implements IDebugRequestHandler, IHotCodeReplaceListener {
```

Comment:
I think the HotCodeReplaceProvider should listen to code change events and perform whatever operation outside the debugger core. It can access the BreakpointManager to manipulate the breakpoints. This can be written outside this handler.

Reply:
Current the breatpointmanager is internally to the core, we can expose it outside via context, but manipulate the breapionts in the HCR will break the encapsulation of the code.  HCR should only care about the HCR work and let other parts do corresponding post work to split the logic.